File: 6 семестр/ПиАПС/Lab 7/Lab 7/Models/BombCellState.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_7.Models
{
    internal class BombCellState : CellState
    {
        public override int Click()
        {
            if (IsMark)
            {
                return -2;
            }
            return GetBombCount();
        }

        public override int GetBombCount()
        {
            return -1;
        }

        public override void Update(IObservable subject)
        {
        }
    }
}

File: 11 semester/Games/LR_5/Source/LR_5/BombCellState.cpp
// Fill out your copyright notice in the Description page of Project Settings.


#include "BombCellState.h"

BombCellState::BombCellState() : CellState()
{
}

BombCellState::~BombCellState()
{
}

int BombCellState::Click()
{
    if (IsMark)
    {
        return -2;
    }
    return GetBombCount();
}

int BombCellState::GetBombCount()
{
    return -1;
}

void BombCellState::Update(IObservable* subject)
{
}
